### ISSUE TITLE
NxCore Data Source - Live Trading

### ISSUE THREAD

**Issue by pablomarin** [QuantConnect/Lean#255]
Guys, 

I want to start contributing to the code. I would like to start by creating the NxCore Data source for live trading.
Can you please tell me everything I need to start the coding process? I know that normally documentation is behind the repository, so i don't want to rely on the documentation 100%. 

Looking forward to your answer.

**Comment by jaredbroad** [comment]
Hi @pablomarin, thankfully this is reasonably simple to do for live-only support:

You would need to implement an IDataQueueHandler, and specify it in the configuration to be used. This type demonstrated in these types: the [Interface](https://github.com/QuantConnect/Lean/blob/master/Common/Interfaces/IDataQueueHandler.cs#L34), and here's an example of it implemented with the [FXCM API](https://github.com/QuantConnect/Lean/blob/master/Brokerages/Fxcm/FxcmBrokerage.DataQueueHandler.cs), and a "[fake datafeed](https://github.com/QuantConnect/Lean/blob/master/Engine/DataFeeds/Queues/FakeDataQueue.cs)" example for local testing.

Basically it is all in the GetNextTicks() method -- simply pipe all the ticks from NxCore into that method and the LEAN engine will filter it and build the bars you need.

PS. I believe NxCore is "all or nothing" -- i.e. they send all the ticks from the market. You can either prefilter these before sending into LEAN based on the "subscription" and Unsubscribe methods, or LEAN will do the filtering for you since we only build bars for ticks in our local subscription list.

**Comment by jaredbroad** [comment]
Hi @pablomarin, closing this for now but if you're still interested we can guide you through setting up NxCore with LEAN. Just ping the ticket and we can reopen it.
